File: src/main/java/com/shuttle/boarding/dto/BoardingConfirmResponse.java
package com.shuttle.boarding.dto;

import java.math.BigDecimal;
import java.time.Instant;

public record BoardingConfirmResponse(
        Long       boardingRecordId,
        Long       tripId,
        String     routeName,
        String     stopName,
        BigDecimal fareAmount,
        String     paymentStatus,
        Instant    boardedAt,
        /** true when this response is from an idempotent replay (not a fresh insert). */
        boolean    alreadyBoarded
) {}

File: lib/features/boarding/data/models/boarding_confirm_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'boarding_confirm_response.g.dart';

/// Response from POST /api/boarding/confirm.
@JsonSerializable()
class BoardingConfirmResponse {
  const BoardingConfirmResponse({
    required this.boardingRecordId,
    required this.tripId,
    required this.routeName,
    required this.stopName,
    this.fareAmount,
    required this.paymentStatus,
    required this.boardedAt,
    required this.alreadyBoarded,
  });

  final int     boardingRecordId;
  final int     tripId;
  final String  routeName;
  final String  stopName;
  final double? fareAmount;
  /// "UNPAID" initially; will become "PAID" after payment.
  final String  paymentStatus;
  final String  boardedAt;   // ISO-8601 string
  /// true if this is an idempotent replay (server already had this record)
  final bool    alreadyBoarded;

  bool get isUnpaid => paymentStatus == 'UNPAID';

  factory BoardingConfirmResponse.fromJson(Map<String, dynamic> json) =>
      _$BoardingConfirmResponseFromJson(json);

  Map<String, dynamic> toJson() => _$BoardingConfirmResponseToJson(this);
}
